docker-compose.yml

```diff
@@ -30,6 +30,44 @@ services:
     volumes:
       - redis-data:/data
 
+  # Based on https://www.elastic.co/guide/en/elasticsearch/reference/current/run-elasticsearch-locally.html
+  # Log in with `elastic` / `insecure-elastic` (or whatever you set the envvar `ES_PASSWORD` to).
+  es:
+    image: docker.elastic.co/elasticsearch/elasticsearch:${ES_VERSION:-8.17.1}
+    environment:
+      - discovery.type=single-node
+      - cluster.name=gpp-search-dev
+      - node.name=es01
+      - node.roles=master,data
+      - ingest.geoip.downloader.enabled=false
+      - ELASTIC_PASSWORD=${ES_PASSWORD:-insecure-elastic}
+      - xpack.security.enabled=true
+      - xpack.security.http.ssl.enabled=false
+      # - xpack.license.self_generated.type=trial
+      - xpack.ml.use_auto_machine_memory_percent=true
+      - ES_JAVA_OPTS=-Xms${ES_HEAP_INIT:-128m} -Xmx${ES_HEAP_MAX:-2g}
+      - cluster.routing.allocation.disk.watermark.low=${ES_DISK_SPACE_REQUIRED:-1gb}
+      - cluster.routing.allocation.disk.watermark.high=${ES_DISK_SPACE_REQUIRED:-1gb}
+      - cluster.routing.allocation.disk.watermark.flood_stage=${ES_DISK_SPACE_REQUIRED:-1gb}
+    volumes:
+      - es-data:/usr/share/elasticsearch/data
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+    healthcheck:
+      test:
+        [
+          "CMD-SHELL",
+          "curl --output /dev/null --silent --head --fail -u elastic:${ES_PASSWORD:-insecure-elastic} http://elasticsearch:9200",
+        ]
+      interval: 5s
+      timeout: 5s
+      retries: 10
+    ports:
+      - "9200:9200"
+      # - "9300:9300"
```

Comment:
Any reason why this is commented out?

Reply:
yeah it's a port that is commonly used by ES which we *currently* don't need but may become relevant in clustering setups